gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/tasks/ValidateSchemaAnnotationTask.java

```diff
@@ -108,12 +119,16 @@ public void validateSchemaAnnotation() throws IOException
       scanHandlersInClassPathJar(f, foundClassNames, scannedClass);
     }
 
-    // skip if no handlers configured or found
-    if (handlerJarPathUrls.size() == 0 || foundClassNames.size() == 0)
+    // For now, every schema annotation handler should be in its own module
+    // if the number of found handlers doesn't match number of configured modules, will throw exception
+    if (foundClassNames.size() != expectedHandlersNumber)
```

Comment:
This wouldn't work if one Jar has more than one handle and another fails to load. A better check would be to ensure atleast one handler per jar dependency.

Ideally, we shouldn't use these heuristics. We should provide a way for users to explicity say which class or package to load and we should error if it cannot be loaded.

Reply:
I thought about this, and I know you are referring to something like 

foundClassNames.size() < expectedHandlersNumber

That Is why I mentioned:
"every schema annotation handler should be in its own module". Can we enforce this rule?

Reply:
It is hard to enforce. I think this change is okay for now, but create a backlog ticket to fix this cleanly. we could discuss during ideas meeting